File: gbt commit 2023.2.8/LoginActivity.java
package com.example.helloword;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.Gravity;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

public class LoginActivity extends AppCompatActivity implements View.OnClickListener {

    private Button mBtLogin;
    private EditText metuser;
    private EditText metpassword;

    @Override
    protected void onCreate(Bundle savedInstanceState){
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_login);

        mBtLogin = findViewById(R.id.btn_1);
        metuser = findViewById(R.id.et_1);
        metpassword = findViewById(R.id.et_2);
        //实现直接跳转方法一

        /*mBtLogin.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = null;
                intent = new Intent(MainActivity.this,FunctionActivity.class);
                startActivity(intent);
            }
        });*/

        //匹配用户名密码的跳转登录


        mBtLogin.setOnClickListener(this);
    }

    public void onClick(View v){

        String username = metuser.getText().toString();
        String password = metpassword.getText().toString();
        //弹出内容设置
        String ok_login="登录成功！";
        String faile_login="账号或密码有误，请重新输入";
        Intent intent = null;

        //对比匹配密码和用户名
        if(username.equals("gbt")&&password.equals("20180701")){
            //用户名和密码匹配成功
            intent = new Intent(LoginActivity.this, HomeActivity.class);
            startActivity(intent);

            Toast.makeText(getApplicationContext(),ok_login,Toast.LENGTH_SHORT).show();
        }
        else{
            //弹出登录失败
            Toast toastcenter = Toast.makeText(getApplicationContext(),faile_login,Toast.LENGTH_SHORT);
            toastcenter.setGravity(Gravity.CENTER,0,0);
            toastcenter.show();
        }

    }

    @Override
    public void onPointerCaptureChanged(boolean hasCapture) {
        super.onPointerCaptureChanged(hasCapture);
    }
}

File: app/src/main/java/com/example/a111111/LoginActivity.kt
package com.example.a111111

import android.annotation.SuppressLint
import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Toast
import com.example.a111111.databinding.ActivityLoginBinding


class LoginActivity : BaseActivity() {
    @SuppressLint("Range")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val binding = ActivityLoginBinding.inflate(layoutInflater)
        setContentView(binding.root)
//对登录界面进行监控
        binding.btnLogin.setOnClickListener {
            val account = binding.etUsername.text.toString()
            val password = binding.etPassword.text.toString()
            val dbHelper = DatabaseHelper(this)
            val db = dbHelper.readableDatabase
            val cursor = db.rawQuery("SELECT * FROM users WHERE username = ?", arrayOf(account))
            if (cursor.moveToFirst()) {
                val username = cursor.getString(cursor.getColumnIndex("username"))
                val pwd = cursor.getString(cursor.getColumnIndex("password"))
                if (password == pwd) {
                    val intent = Intent(this, Switch11Activity::class.java)
                    startActivity(intent)
                    finish()
                } else {
                    Toast.makeText(this, "密码错误", Toast.LENGTH_SHORT).show()
                }
            } else {
                Toast.makeText(this, "用户不存在", Toast.LENGTH_SHORT).show()
            }
            cursor.close()
            db.close()
        }
//对注册按钮进行监控
        binding.btnZhu.setOnClickListener{
            val intent = Intent(this, RegisterActivity::class.java)
            startActivity(intent)
        }
    }
}